docs/resources/vm_qemu.md

```diff
@@ -69,108 +61,178 @@ EOF
+|`macaddr`|`str`||Override the randomly generated MAC Address for the VM.|
+|`bridge`|`str`|`"nat"`|Bridge to which the network device should be attached. The Proxmox VE standard bridge is called `vmbr0`.|
+|`tag`|`int`|`-1`|The VLAN tag to apply to packets on this device. `-1` disables VLAN tagging.|
+|`firewall`|`bool`|`false`|Whether to enable the Proxmox firewall on this network device.|
+|`rate`|`int`|`0`|Network device rate limit in mbps (megabytes per second) as floating point number. Set to `0` to disable rate limiting.|
+|`queues`|`int`|`1`|Number of packet queues to be used on the device. Requires `virtio` model to have an effect.|
+|`link_down`|`bool`|`false`|Whether this interface should be disconnected (like pulling the plug).|
+
+### Disk Block
+
+The `disk` block is used to configure the disk devices. It may be specified multiple times. The order in which the blocks are specified and the disk device type determines the ID for each disk device. Take the following for example:
+
+```hcl
+resource "proxmox_vm_qemu" "resource-name" {
+    //<arguments ommitted for brevity...>
+
+    disk { // This disk will become scsi0
+        type = "scsi"
+
+        //<arguments ommitted for brevity...>
+    }
+
+    disk { // This disk will become ide0
+        type = "ide"
+
+        //<arguments ommitted for brevity...>
+    }
+
+    disk { // This disk will become scsi1
+        type = "scsi"
+
+        //<arguments ommitted for brevity...>
+    }
+
+    disk { // This disk will become sata0
+        type = "sata"
+
+        //<arguments ommitted for brevity...>
+    }
+}
+```
+
+See the [docs about disks](https://pve.proxmox.com/pve-docs/chapter-qm.html#qm_hard_disk) for more details.
+
+|Argument|Type|Default Value|Description|
+|--------|----|-------------|-----------|
+|`type`|`str`||**Required** The type of disk device to add. Options: `ide`, `sata`, `scsi`, `virtio`.|
+|`storage`|`str`||**Required** The name of the storage pool on which to store the disk.|
+|`size`|`str`||**Required** The size of the created disk, format must match the regex `\d+[GMK]`, where G, M, and K represent Gigabytes, Megabytes, and Kilobytes respectively.|
+|`format`|`str`|`"raw"`|The drive’s backing file’s data format.|
+|`cache`|`str`|`"none"`|The drive’s cache mode. Options: `directsync`, `none`, `unsafe`, `writeback`, `writethrough`|
+|`backup`|`bool`|`false`|Whether the drive should be included when making backups.|
+|`iothread`|`bool`|`false`|Whether to use iothreads for this drive. Only effective with a disk of type `virtio`, or `scsi` when the the emulated controller type (`scsihw` top level block argument) is `virtio-scsi-single`.|
```

Comment:
this parameter(maybe some others) should be an integer, ad discovered in issue #283 

![api](https://user-images.githubusercontent.com/807457/103998776-9024a200-519c-11eb-8b4c-bc881db356c6.png)

Reply:
I agree, on the backend `iothread` (and a couple other parameters,) should be `int`s. However they are currently defined as `bool`s in the resource schema, so if you were to supply an `int` in the terraform configuration, terraform would return a validation error.

https://github.com/Telmate/terraform-provider-proxmox/blob/944cd338944b838c842263e42c57364b97be9c82/proxmox/resource_vm_qemu.go#L385-L389

A separate PR is needed to either change the schema to `int`s, or they could remain as `bool`s (since that's effectively what `1` and `0` represent anyway,) and the provider could be changed to cast the `bool`s to an `int` before passing the parameters to the PVE API.

Reply:
Correct, I've made the PR to change types of
- iothread
- backup
- replicate
- ssd

from bool to int.
I would stick to API types instead of doing hacks in provider, but feel free to modify if I forgot somthing or if my approach is wrong :)